devicetypes/smartthings/thirdreality-waterleak-sensor.src/thirdreality-waterleak-sensor.groovy

```diff
@@ -0,0 +1,116 @@
+/**
+ *  ThirdReality WaterLeak Sensor
+ *
+ *  Copyright 2021 THIRDREALITY
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+    definition (name: "ThirdReality WaterLeak Sensor", namespace: "smartthings", author: "THIRDREALITY", cstHandler: true) {
+        capability "Battery"
+        capability "Switch"
+        capability "Water Sensor"
+        capability "Refresh"
+        capability "Configuration"
+
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0001,0500", outClusters: "0006,0019", manufacturer:"Third Reality, Inc", model:"3RWS18BZ", deviceJoinName: "ThirdReality Water Leak Sensor"		//ThirdReality WaterLeak Sensor
+    }
+
+    simulator {
+        // When simulating, define status and reply messages here
+    }
+
+    tiles {		//Seems no use
+        // define your main and details tiles here
+        main("water")
+        details(["water", "battery", "switch"])
+    }
+}
+
+def getBIND_CLUSTER() {0x8021}
+
+// parse events into attributes
+def parse(String description) {
+    log.trace "[parse] Parsing '${description}'"
+    def resMap = [:]
+
+    if (description?.startsWith("zone status")) {
+        resMap = createEvent(name: "water", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "wet" : "dry")
+        if (getDataValue("onoff_state") != "on") {
+            sendEvent(name: "switch", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "on":"off")
+        }
+    } else if (description?.startsWith("on/off")) {
+        resMap = zigbee.getEvent(description)
+        updateDataValue("onoff_state", resMap.value)
```

Comment:
you could just be using `device.currentState("switch")` here to get the current value of the Switch capability

Reply:
Truely, thanks! This way is better than mine. :D

Reply:
Sorry, I have tried. But this is not what I want. Because the Sensor icon is activated under "On" switch state, and Zigbee OnOff cmd is used to control a buzzer on WaterLeak Sensor, I wrote a logic like:
if under On State: if rcv Wet report, Icon Active(Switch, On)
                             if rcv Dry report, Icon Active(Switch, On)
if under Off State: if rcv Wet report, Icon Active(Switch, On)
                             if rcv Dry report, Icon Inactive(Switch, Off)
See above, I need 3 variables to finish the logic:
Global On/Off State (onoff_state) is used to control the buzzer on WaterLeak Sensor. Users can touch the App to On to make the WaterLeak buzzing.
Wet/Dry State (water) is used to describe the current state of water sensor.
Switch On/Off(switch) is used to activate the Icon in App.

Maybe the name "onoff_state" is confusing. So I changed the name into "buzzing_state"